src/adera/Machines/Rocket.cpp

```diff
@@ -108,18 +107,18 @@ void SysMachineRocket::update_physics()
         else
         {
             // rocket's rigid body not set yet
-            auto body = m_physics.find_rigidbody_ancestor(ent);
+            auto [bodyEnt, pBody]
+                    = SysPhysics_t::find_rigidbody_ancestor(m_scene, ent);
 
-            if (body.second == nullptr)
+            if (pBody == nullptr)
             {
                 std::cout << "no rigid body!\n";
                 continue;
             }
 
-            machine.m_rigidBody = body.first;
-            compRb = body.second;
-            compTf = m_scene.get_registry()
-                    .try_get<ACompTransform>(body.first);
+            machine.m_rigidBody = bodyEnt;
+            compRb = pBody;
```

Comment:
If you're not accepting it by value, you probably want to use std::move().

Reply:
bodyEnt is an entity, it is really small.

Reply:
but pBody is a shared_pointer, isn't it?

Reply:
Or is it a raw pointer?

Reply:
Yes, it is a raw pointer to ACompNwtBody*. The `find_rigidbody_ancestor` function will return nullptr if it doesn't find a ACompNwtBody. Function is defined here:

https://github.com/TheOpenSpaceProgram/osp-magnum/blob/2ccf99891dd4611ebac6b6a010b1f5c1d5056a99/src/osp/Active/SysNewton.cpp#L361